### DISCUSSION TITLE
How To Configure Gulp For Different Repos

### DISCUSSION THREAD

**Discussion by DrAnonymous** [forkineye/ESPixelStick#448] [Q&A] [answered]
I cloned the repo locally and got Gulp setup and was able to eventually compile everything. I then forked main and cloned that, but when I went to run Gulp I got this error:  

```
Local modules not found in ~/<path to my clone>/ESPixelStick  
Try running: npm install  
```

Which part of the steps listed on [the gulp setup](https://github.com/forkineye/ESPixelStick/blob/main/html/README.md#how-to-setup-and-use-gulp) are unique to each cloned directory tree?

With all the automation that platform.io seems to have, is Gulp really needed to process the webpages? Looking over the gulp script it appears to do the following:  
- Remove all whitespace from html/js files
- Compress directories into a gzip
- Copy compressed files to data directory  

Is that an oversimplification?

**Comment by forkineye** [comment]
It concatenates them as well. I've been working on a complete rewrite of the UI in reactjs, so everything about the frontend build (gulp, etc..) will be changing drastically in the future.  npm install will drop everything into the node_modules directory of where package.json is.

**Reply by DrAnonymous** [reply]
Thanks for the info.

**Comment by DrAnonymous** [comment, marked as answer]
I was able to get things working again by running: 

```
npm install   
gulp
```

Running the original steps, after having done them with the prior clone, didn't work. I'm sure its ignorance on my part.
